ui/core/components/results_viewer.tsx

```diff
@@ -127,4 +133,20 @@ export class ResultsViewer extends Component {
 		this.contentElem.style.display = 'block';
 		this.pendingElem.style.display = 'none';
 	}
+
+	addAbortButton(abortClicked: (event: MouseEvent) => void) {
+		this.buttonElem.innerHTML = '';
+		this.buttonElem.style.display = 'block';
+		this.buttonElem.appendChild(
+			<button className="sim-abort-button" onclick={abortClicked}>
+				<i className="fa fa-times fa-lg me-1" />
+				Stop
+			</button>,
+		);
+	}
+
+	removeAbortButton() {
+		this.buttonElem.innerHTML = '';
```

Comment:
This can be  `this.buttonElemen.replaceChildren()`, We are trying to get rid of all `innerHTML` things